examples/pose_graph.py

```diff
@@ -12,37 +12,26 @@
 
 file_path = "datasets/tinyGrid3D.g2o"
 
-num_verts, verts, edges = theg.pose_graph.read_3D_g2o_file(file_path)
-
-objective = th.Objective(torch.float64)
-
-for edge in edges:
-    cost_function = theg.RelativePoseError(
-        verts[edge.i], verts[edge.j], edge.relative_pose, edge.weight
-    )
-    objective.add(cost_function)
-
-torch.manual_seed(1)
-
-file_path = "datasets/tinyGrid3D.g2o"
-
 th.SO3.SO3_EPS = 1e-6
 
-num_verts, verts, edges = theg.pose_graph.read_3D_g2o_file(file_path)
+num_verts, verts, edges = theg.pose_graph.read_3D_g2o_file(
+    file_path, dtype=torch.float32
+)
 
-objective = th.Objective(torch.float64)
+objective = th.Objective(torch.float32)
 
 for edge in edges:
-    cost_func = theg.RelativePoseError(
-        verts[edge.i], verts[edge.j], edge.relative_pose, edge.weight
+    cost_func = th.eb.Between(
+        verts[edge.i], verts[edge.j], edge.weight, edge.relative_pose
     )
     objective.add(cost_func)
 
-objective.add(
-    theg.PosePriorError(
-        pose=verts[0], pose_prior=verts[0].copy(new_name=verts[0].name + "PRIOR")
-    )
+pose_prior = th.eb.VariableDifference(
```

Comment:
I'm confused about this cost function. Is this a constraint to make the initial pose exactly some value? Why is the weight so small?

Reply:
Correct. The only reason we introduce `pose_prior` is to make the linear system not ill-conditioned. A too large weight will restrict the search space and yield slow convergence.

Reply:
`GTSAM` also has similar factors [[link](https://github.com/borglab/gtsam/blob/0850e89b0e6148d16f00b7741d83b6325ada977d/examples/Pose2SLAMExample.cpp#L73)].

Reply:
If I understand that code, they are anchoring that variable to a zero/origin value, which is not the case here? Here the target is set to whatever value `verts[0]` has at this point.

Reply:
I don't think the `target` value matters. In fact, the `target` value can be anything. In PGO problems, we mainly consider the relative pose errors, which, however, has no constraints on the absolute pose. Thus, we introduce pose prior add absolute pose constraint.

The reason why I use `verts[0]` is that it is closed to the initial guess.

Reply:
I see. Thanks for the explanation!